packages/flutter/lib/src/widgets/_web_image_web.dart

```diff
@@ -0,0 +1,241 @@
+import '../painting/_network_image_web.dart';
+import '../web.dart' as web;
+
+/// Returns [true] if the bytes at [url] can be fetched.
+///
+/// The bytes may be unable to be fetched if they aren't from the same origin
+/// and the server hosting them does not allow cross-origin requests.
+Future<bool> checkIfImageBytesCanBeFetched(String url) async {
+  // First check if url is same origin.
+  final Uri uri = Uri.parse(url);
+  if (uri.origin == web.window.origin) {
+    return true;
+  }
+
+  final Completer<web.XMLHttpRequest> completer =
+      Completer<web.XMLHttpRequest>();
+  final web.XMLHttpRequest request = httpRequestFactory();
+
+  request.open('HEAD', url, true);
+
+  request.addEventListener(
+      'load',
+      (web.Event e) {
+        final int status = request.status;
+        final bool accepted = status >= 200 && status < 300;
+        final bool fileUri = status == 0; // file:// URIs have status of 0.
+        final bool notModified = status == 304;
+        final bool unknownRedirect = status > 307 && status < 400;
+        final bool success =
+            accepted || fileUri || notModified || unknownRedirect;
+
+        if (success) {
+          completer.complete(request);
+        } else {
+          completer.completeError(e);
+        }
+      }.toJS);
+
+  request.addEventListener(
+      'error', ((JSObject e) => completer.completeError(e)).toJS);
+
+  request.send();
+
+  return completer.future.then((_) {
+    return true;
+  }, onError: (_) {
+    return false;
+  });
+}
+
+/// Returns a widget which displays the [src] in an <img> tag.
+Widget createImgElementWidget(String src,
+        {Key? key,
+        ImageLoadingBuilder? loadingBuilder,
+        ImageFrameBuilder? frameBuilder,
+        ImageErrorWidgetBuilder? errorBuilder}) =>
+    _ImgElementImage(
+      src,
+      key: key,
+      loadingBuilder: loadingBuilder,
+      frameBuilder: frameBuilder,
+      errorBuilder: errorBuilder,
+    );
+
+class _ImgElementImage extends StatefulWidget {
+  const _ImgElementImage(
+    this.src, {
+    super.key,
+    this.loadingBuilder,
+    this.frameBuilder,
+    this.errorBuilder,
+  });
+
+  final String src;
+  final ImageLoadingBuilder? loadingBuilder;
+  final ImageFrameBuilder? frameBuilder;
+  final ImageErrorWidgetBuilder? errorBuilder;
+
+  @override
+  State<StatefulWidget> createState() => _ImgElementImageState();
+}
+
+class _ImgElementImageState extends State<_ImgElementImage> {
+  _ImgElementImageState() {
+    if (_registeredViewType == null) {
+      _register();
+    }
+  }
+
+  // Keeps track if this widget has already registered its view factories.
+  static String? _registeredViewType;
+
+  late _CachedImageInfo _cachedImage;
+
+  void _onCachedImageStateChange() {
+    // Rebuild when the image state changes.
+    setState(() {});
+  }
+
+  static const String _viewType = 'Browser__ImageElementType__';
+
+  static void _register() {
+    assert(_registeredViewType == null);
```

Comment:
`_registeredViewType` isn't ever set to anything else than null?

Reply:
Oops, I deleted the line to set it when doing some refactoring. I restored it.